spark/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -147,6 +146,17 @@ case class OptimizeTableCommand(
       throw DeltaErrors.notADeltaTableException(table.deltaLog.dataPath.toString)
     }
 
+    // Validate that the preview is enabled if we are optimizing a clustered table.
+    ClusteredTableUtils.validatePreviewEnabled(txn.snapshot.protocol)
```

Comment:
put this under `if (ClusteredTableUtils.isSupported(txn.protocol))`?

Reply:
yea, but why waste calling it for the non-clustered table scenario?

Reply:
in terms of calls, it may be fine (you need to run `isSupported` at least once anyway`. But I think it makes more sense to put the validation under the below `if` along with other checks.

Reply:
Yeah, makes sense. I moved to inside the if below.